GNUmakefile

```diff
@@ -55,6 +55,12 @@ real_tests: bin
 sanity_check: tools/sanity_check.sh
 	@./$<
 
+.PHONY: git-setup
+git-setup:
+	@printf '#!/bin/sh\n[ -z "$$(make sanity_check)" ]\n' > \
+.git/hooks/pre-commit
```

Comment:
What if someone has its own pre-commit file already?

Reply:
Well, perhaps we can expect that person not to run `make git-setup`.

Reply:
Or that person shall know how to call sanity check from his/her own hook.

Reply:
I agree with @fblanqui , as it is you erase the file if it exists. You should add a check before and print a message if it already exists.

Reply:
OK. In first place I wanted to have a short instruction, but you're both right. I'll fix it.